### PR TITLE
add logs for entry missing

### PR DESCRIPTION
**Pull request by BusyJay**
### What is changed and how it works?
Issue Number: Ref #12664 

What's Changed:

```commit-message
...So we can know what is actually happening.
```

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by zhangjinpeng1987 — APPROVED** [review]
LGTM

File: components/raftstore/src/store/fsm/store.rs [hunk 1]
```diff
@@ -2035,11 +2035,20 @@ impl<'a, EK: KvEngine, ER: RaftEngine, T: Transport> StoreFsmDelegate<'a, EK, ER
 
         let mut is_overlapped = false;
         let mut regions_to_destroy = vec![];
-        for (_, id) in meta.region_ranges.range((
+        for (key, id) in meta.region_ranges.range((
             Excluded(data_key(msg.get_start_key())),
             Unbounded::<Vec<u8>>,
         )) {
-            let exist_region = &meta.regions[id];
+            let exist_region = match meta.regions.get(id) {
```
Comment (5kbpers) [inline components/raftstore/src/store/fsm/store.rs:2042]: We have many other places using `meta.regions[..]`. How about extracting to a function for it?

Comment (BusyJay) [reply components/raftstore/src/store/fsm/store.rs:2042]: Thought about it, but those places were less frequently visited and context everywhere might be too verbose.

**Review by 5kbpers — APPROVED** [review]

**Review by BornChanger — APPROVED** [review]

**Pull request merged**